AppLovin MAX Demo App - ObjC/AppLovin MAX Demo App - ObjC/MAX/Native Ads/ALMAXManualNativeAdViewController.m

```diff
@@ -0,0 +1,124 @@
+//
+//  ALMAXTemplateNativeAdViewController.m
+//  AppLovin MAX Demo App - ObjC
+//
+//  Created by Billy Hu on 1/20/22.
+//  Copyright © 2022 AppLovin Corporation. All rights reserved.
+//
+
+#import "ALMAXManualNativeAdViewController.h"
+#import "ALBaseAdViewController.h"
+#import <Adjust/Adjust.h>
+#import <AppLovinSDK/AppLovinSDK.h>
+
+@interface ALMAXManualNativeAdViewController()<MANativeAdDelegate, MAAdRevenueDelegate>
+@property (nonatomic, weak) IBOutlet UIView *nativeAdContainerView;
+
+@property (nonatomic, strong) MANativeAdLoader *nativeAdLoader;
+@property (nonatomic, strong) MANativeAdView *nativeAdView;
+@property (nonatomic, strong, nullable) MAAd *nativeAd;
+
+@end
+
+@implementation ALMAXManualNativeAdViewController
+
+#pragma mark - View Lifecycle
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    UINib *nativeAdViewNib = [UINib nibWithNibName: @"NativeCustomAdView" bundle: NSBundle.mainBundle];
+    
+    self.nativeAdView = [nativeAdViewNib instantiateWithOwner: nil options: nil].firstObject;
+    
+    MANativeAdViewBinder *binder = [[MANativeAdViewBinder alloc] initWithBuilderBlock:^(MANativeAdViewBinderBuilder *builder) {
+        builder.titleLabelTag = 1001;
+        builder.advertiserLabelTag = 1002;
+        builder.bodyLabelTag = 1003;
+        builder.iconImageViewTag = 1004;
+        builder.optionsContentViewTag = 1005;
+        builder.mediaContentViewTag = 1006;
+        builder.callToActionButtonTag = 1007;
+    }];
+    [self.nativeAdView bindViewsWithAdViewBinder: binder];
+    
+    self.nativeAdLoader = [[MANativeAdLoader alloc] initWithAdUnitIdentifier: @"YOUR_AD_UNIT"];
+    self.nativeAdLoader.nativeAdDelegate = self;
+    self.nativeAdLoader.revenueDelegate = self;
+}
+
+- (void)dealloc
+{
+    [self cleanUpAdIfNeeded];
+    
+    self.nativeAdLoader.nativeAdDelegate = nil;
+    self.nativeAdLoader.revenueDelegate = nil;
+}
+
+#pragma mark - IB Actions
+
+- (IBAction)showAd
+{
+    [self cleanUpAdIfNeeded];
+    
+    [self.nativeAdLoader loadAdIntoAdView: self.nativeAdView];
+}
+
+- (void)cleanUpAdIfNeeded
```

Comment:
nit: I would move this above the `#pragma mark - IB Actions` since this is unrelated to IBActions